src/webui/webapplication.cpp

```diff
@@ -774,6 +776,26 @@ void WebApplication::action_command_bottomPrio()
     BitTorrent::Session::instance()->bottomTorrentsPriority(hashes);
 }
 
+void WebApplication::action_command_rename()
+{
+    CHECK_URI(0);
+    CHECK_PARAMETERS("hash" << "name");
+
+    QString hash = request().posts["hash"];
+    QString name = request().posts["name"].trimmed();
+
+    BitTorrent::TorrentHandle *const torrent = BitTorrent::Session::instance()->findTorrent(hash);
+    if (torrent && !name.isEmpty()) {
+        name.replace(QRegularExpression("\r?\n|\r"), " ");
```

Comment:
I think `torrent->setName(name);` will handle all invalid chars, so maybe this isn't really needed.
Might need some testing.